pygmt/src/timestamp.py

```diff
@@ -0,0 +1,102 @@
+"""
+timestamp - Plot the GMT timestamp logo.
+"""
+from packaging.version import Version
+from pygmt import __gmt_version__
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_string, is_nonstr_iter
+
+__doctest_skip__ = ["timestamp"]
+
+
+def timestamp(
+    self,
+    text=None,
+    label=None,
+    justification="BL",
+    offset=("-54p", "-54p"),
+    font="Helvetica,black",
+    timefmt="%Y %b %d %H:%M:%S",
+):
+    r"""
+    Plot the GMT timestamp logo.
+
+    Parameters
+    ----------
+    text : None or str
+        If ``None``, the current UNIX timestamp is shown in the GMT timestamp
+        logo. Set this parameter to replace the UNIX timestamp with a
+        custom text string instead. The text must be less than 64 characters.
+        *Requires GMT>=6.5.0*.
+    label : None or str
+        The text string shown after the GMT timestamp logo.
+    justification : str
+        Justification of the timestamp. The *justification* is a two-character
+        code that is a combination of a horizontal (**L**\ (eft),
+        **C**\ (enter), or **R**\ (ight)) and a vertical (**T**\ (op),
+        **M**\ (iddle), or **B**\ (ottom)) code.
+    offset : str or tuple
+        *offset* or (*offset_x*, *offset_y*).
+        Offset the anchor point of the timestamp by *offset_x* and *offset_y*.
+        If a single value *offset* is given, *offset_y* = *offset_x* =
+        *offset*.
+    font : str
+        Font of the timestamp and the optional label. The parameter can't
+        change the font color for GMT<=6.4.0.
```

Comment:
Should a GMTInvalidInput error be raised when users pass `font` as a parameter with GMT 6.3.0? Or is the error clear enough from GMT?

Reply:
The `font` parameter works for any GMT versions. 

For GMT>6.4.0, it can change both the **font ID** and **font color**, but for GMT<=6.4.0, the font color is always black so it can only change the font ID.

Reply:
Ah ok, maybe reword it into something like this?
```suggestion
        Font of the timestamp and the optional label. The parameter can't
        change the font color for GMT<=6.4.0, only the font label.
```

Reply:
I'm afraid "font label" is incorrect here. It should be "font ID" or "font name".

Reply:
Owh, you mean like font style (e.g. Helvetica)? Feel free to edit the suggested changes.

Reply:
Added "only the font ID" in 007964375e091a93c8e7c63c840fb3affc2a8e01.